src/main/scala/firrtl/transforms/Dedup.scala

```diff
@@ -165,7 +165,7 @@ class DedupModules extends Transform with DependencyAPIMigration with PreservesA
     }.toMap
 
     // get the ordered set of instances a module, includes new Deduped modules
-    val getChildrenInstances = (mod: String) => {
+    val getChildrenInstances = {
       val childrenMap = instanceGraph.getChildrenInstances
       val newModsMap: Map[String, mutable.LinkedHashSet[WDefInstance]] = dedupMap.map {
```

Comment:
I see that your way prevents re-computation, but since these are constant across the calls to `getChildrenInstances` there any reason not to just pull these data structures out of this anonymous scope, make them `val`s scoped to `run` and then make `getChildrenInstances` itself a one-liner def?

Reply:
I could do that. I did it this way partially to minimize the diff, although also partially that I kind of like the pattern of using anonymous scopes to contain vals that are only used inside of it, I find it self-documenting that these aren't used later. I'm happy to change it if you think it's more canonical or clear though.

Reply:
The return value is a function returning a mutable set, which means that it actually requires some detailed reading to know that the fact that data structures scoped to this anonymous closure are not later mutated.

I certainly don't object to just merging it as-is, but I think that the "anonymous closure to hide random local variables" is maybe a little less accessible. The subtle difference of an immediately-evaluated closure in a pure expression position vs. a traditional `val` function closure was what obscured the repetitive evaluation.

This is all totally a silly preference thing. I assumed you did it just to minimized the diff, but if you like it you should keep it :)